sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpClient.java

```diff
@@ -122,33 +147,26 @@ private java.net.http.HttpRequest toJdkHttpRequest(HttpRequest request, Context
      * @return the java major version
      */
     private static int getJavaVersion() {
-        // java.version format:
-        // 8 and lower: 1.7, 1.8.0
-        // 9 and above: 12, 14.1.1
-        String version = System.getProperty("java.version");
-        if (CoreUtils.isNullOrEmpty(version)) {
-            throw LOGGER.logExceptionAsError(new RuntimeException("Can't find 'java.version' system property."));
-        }
-        if (version.startsWith("1.")) {
-            if (version.length() < 3) {
-                throw LOGGER.logExceptionAsError(new RuntimeException("Can't parse 'java.version':" + version));
-            }
-            try {
-                return Integer.parseInt(version.substring(2, 3));
-            } catch (Exception t) {
-                throw LOGGER.logExceptionAsError(new RuntimeException("Can't parse 'java.version':" + version, t));
-            }
-        } else {
-            int idx = version.indexOf(".");
+        return Runtime.version().feature();
+    }
 
-            if (idx == -1) {
-                return Integer.parseInt(version);
-            }
-            try {
-                return Integer.parseInt(version.substring(0, idx));
-            } catch (Exception t) {
-                throw LOGGER.logExceptionAsError(new RuntimeException("Can't parse 'java.version':" + version, t));
-            }
-        }
+    /**
+     * Gets the response body handler based on whether a read timeout is configured.
+     * <p>
+     * When a read timeout is configured our custom handler is used that tracks the time taken between each read
+     * operation to pull the body from the network. If a timeout isn't configured the built-in JDK handler is used.
+     *
+     * @param readTimeout The configured read timeout.
+     * @param jdkBodyHandler The JDK body handler to use when no read timeout is configured.
+     * @param timeoutSubscriber The supplier for the custom body subscriber to use when a read timeout is configured.
+     * @return The response body handler to use.
+     * @param <T> The type of the response body.
+     */
+    private static <T> java.net.http.HttpResponse.BodyHandler<T> getResponseHandler(Duration readTimeout,
+        Supplier<java.net.http.HttpResponse.BodyHandler<T>> jdkBodyHandler,
+        Function<Long, java.net.http.HttpResponse.BodySubscriber<T>> timeoutSubscriber) {
+        return (readTimeout != null && !readTimeout.isNegative() && !readTimeout.isZero())
```

Comment:
can we validate/normalize timeout centrally  in azure-core ? In `HttpClientOptions`,  `HttpClientBuilder` or somewhere else so we don't need to repeat it in every clients and keep things in sync?

Reply:
Yeah, I think this can get cleaned up. Will do that after this round of CI runs to make sure there aren't more fixes to be done.

Reply:
Created #39456